apps/api/src/domain/models/audio-item/commands/translate-audio-item-name/translate-audio-item.command.integration.spec.ts

```diff
@@ -0,0 +1,124 @@
+            new MultilingualTextItem({
+                text: 'original name of the audio item',
+                role: MultilingualTextItemRole.original,
+                languageCode: LanguageCode.Chilcotin,
+            }),
+        ],
+    }),
+});
+
+const englishName = 'audio item name translated to English';
+
+const validPayload: TranslateAudioItemName = {
+    aggregateCompositeIdentifier: {
+        type: AggregateType.audioItem,
+        id: existingAudioItem.id,
+    },
+    languageCode: LanguageCode.English,
+    text: englishName,
+};
+
+const validCommandFSA = {
+    type: commandType,
+    payload: validPayload,
+};
+
+const buildValidCommandFSA = (): FluxStandardAction<DTO<TranslateAudioItemName>> => validCommandFSA;
+
+describe(commandType, () => {
+    let app: INestApplication;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let idManager: IIdManager;
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }).catch((error) => {
+                throw error;
+            }));
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCommandSuccess(commandAssertionDependencies, {
+                systemUserId: dummySystemUserId,
+                buildValidCommandFSA,
+                initialState: new DeluxeInMemoryStore({
+                    [AggregateType.audioItem]: [existingAudioItem],
+                }).fetchFullSnapshotInLegacyFormat(),
+                checkStateOnSuccess: async ({
+                    aggregateCompositeIdentifier: { id: audioItemListId },
+                }: TranslateAudioItemName) => {
+                    const audioItemSearchResult = await testRepositoryProvider
+                        .forResource(ResourceType.audioItem)
+                        .fetchById(audioItemListId);
+
+                    expect(audioItemSearchResult).not.toBe(NotFound);
+
+                    const audioItem = audioItemSearchResult as AudioItem;
+
+                    const englishAudioItemNameTranslation = audioItem.name.getTranslation(
+                        LanguageCode.English
+                    );
+
+                    expect((englishAudioItemNameTranslation as MultilingualTextItem).text).toBe(
+                        englishName
+                    );
+                },
+            });
+        });
+    });
```

Comment:
Be sure to add a
```ts
describe(`when the command is invalid`,...)
```

## Invalid Test Cases
- When the audio item with the given ID does not exist
- When the audio item's name has already been translated into this language
- When the audio item was named in the given language to start with (original item's language code is the translation language code)
- fuzz test
- when `payload.aggregateCompositeIdentifier` is not `AggregateType.audioItem` (loop over all other aggregate types as in similar tests)